source/Playnite/Scripting/PowerShell/PowerShell.cs

```diff
@@ -128,12 +138,42 @@ public object GetVariable(string name)
             return runspace.SessionStateProxy.GetVariable(name);
         }
 
-        public bool GetFunctionExits(string name)
+        public CommandInfo GetFunction(string name)
         {
-            using (var pipe = runspace.CreatePipeline($"Get-Command {name} -EA 0"))
+            var command = powershell
+                .AddCommand("Get-Command")
+                .AddArgument(name)
+                .AddParameter("PassThru")
+                .AddParameter("ErrorAction", ActionPreference.SilentlyContinue)
+                .Invoke<CommandInfo>().FirstOrDefault();
+            powershell.Streams.ClearStreams();
+            powershell.Commands.Clear();
+            return command;
+        }
+
+        public object InvokeFunction(string name, List<object> arguments)
+        {
+            var command = GetFunction(name);
+            powershell.AddCommand(command);
+            // Don't pass arguments to commands not expecting parameters
+            if (command.Parameters.Count != 0)

```

Comment:
This is the one change I'm most concerned with breaking compatibility.

Explicitly invoking commands with this API actually enforces that the number of arguments match the command being invoked. That is, providing 2 arguments to a function expecting 0 will cause an error. This is contrary to how PowerShell usually behaves silently ignores the extra arguments.

This is a problem because many extensions ignore arguments passed to them that they don't need. As a workaround, I decided to special case commands expecting exactly 0 arguments and don't pass any if so.

Reply:
I was trying to remove any time strings were eval'ed. Previously the function name and it's args were executed as a string. Maybe there's another API that avoids the parameters problem, but I didn't find it.

Reply:
Would you like me to remove this `if (command.Parameters.Count != 0)` test and require that exported functions accept the correct number of arguments?

Reply:
Yes remove it. Exported functions must have correct number of arguments. This should not be an issue in future because I'll be wrapping all arguments under single object (for better future proofing).